src/rdkafka_fetcher.c

```diff
@@ -611,15 +606,19 @@ rd_kafka_fetch_reply_handle_partition(rd_kafka_broker_t *rkb,
         if (unlikely(err))
                 rd_kafka_toppar_fetch_backoff(rkb, rktp, err);
 
-        rd_kafka_toppar_destroy(rktp); /*from get()*/
-
-        return RD_KAFKA_RESP_ERR_NO_ERROR;
+        goto no_err;
 
 err_parse:
         if (rktp)
                 rd_kafka_toppar_destroy(rktp); /*from get()*/
-
+        rd_kafka_buf_skip_tags(rkbuf);
```

Comment:
this isn't needed because it doesn't check if it has read the whole buffer in case of failure. Add a conditional `rd_kafka_aborted_txns_destroy` here too.